src/main/scala/viper/silver/reporter/Message.scala

```diff
@@ -269,11 +270,31 @@ case class CopyrightReport(override val text: String) extends SimpleMessage(text
 }
 
 case class MissingDependencyReport(override val text: String) extends SimpleMessage(text) {
-  override val name = "missing_dependency_report"
+  override val name: String = "missing_dependency_report"
 }
 
 // FIXME: for debug purposes only: a pong message can be reported to indicate
 // FIXME: that the verification backend is alive.
 case class PongMessage(override val text: String) extends SimpleMessage(text) {
   override val name: String = "dbg__pong"
 }
+
+case class QuantifierInstantiationsMessage(quantifier: String, instantiations: Int,
+                                           max_gen: Int, max_cost: Int) extends Message {
+  override lazy val toString: String = s"quantifier_instantiations_message(" +
+    s"quantifier=${quantifier}, instantiations=${instantiations.toString}," +
+    s"max_gen=${max_gen}, max_cost=${max_cost})"
+  override val name: String = "quantifier_instantiations_message"
+}
+
+case class QuantifierChosenTriggersMessage(quantifier: QuantifiedExp, triggers: Seq[Trigger]) extends Message {
+  override lazy val toString: String = s"quantifier_chosen_triggers_message(type=$quant_type, quantifier=${quantifier.toString}, triggers=$triggers_string)"
+  override val name: String = "quantifier_chosen_triggers_message"
+  lazy val triggers_string: String = triggers.map((trigger) => trigger.exps.mkString("{", ", ", "}")).mkString("[", ", ", "]")
+  val quant_type: String = quantifier.getClass.getSimpleName
+}
+
+case class VerificationTerminationMessage() extends Message {
+  override lazy val toString: String = s"verification_termination_message"
```

Comment:
Don't use a `lazy val` here: it incurs synchronisation overhead for most likely no performance gain, since the string is a literal. Drop the `s"..."` prefix as well, if no string interpolation is intended.